pkg/util/util.go

```diff
@@ -74,3 +78,47 @@ func ErrList(e []error) error {
 	}
 	return nil
 }
+
+func FetchClusterUID(namespaces v1core.NamespaceController) (string, error) {
+	kubesystemNamespace, err := namespaces.Get("kube-system", k8sv1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	return string(kubesystemNamespace.UID), nil
+}
+
+// Define the GroupVersionResource for CRDs
+var crdGVR = schema.GroupVersionResource{
+	Group:    "apiextensions.k8s.io",
+	Version:  "v1",
+	Resource: "customresourcedefinitions",
+}
+
+func getCRDDefinition(dynamicClient apiextensionsClientSetv1.ApiextensionsV1Interface, crdName string) (*apiextensionsv1.CustomResourceDefinition, error) {
+	crd, err := dynamicClient.CustomResourceDefinitions().Get(context.TODO(), crdName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return crd, nil
+}
+
+func VerifyBackupCrdHasClusterStatus(client apiextensionsClientSetv1.ApiextensionsV1Interface) bool {
+	crdName := "backups.resources.cattle.io"
+
+	crd, err := getCRDDefinition(client, crdName)
+	if err != nil {
+		logrus.Infof("Error fetching CRD: %v", err)
+		return false
+	}
+
+	// Inspect the status schema, for example
+	_, found := crd.Spec.Versions[0].Schema.OpenAPIV3Schema.Properties["status"].Properties["originCluster"]
+	if found {
+		logrus.Debugf("Status schema contains `originCluster` on CRD `%s`.\n", crdName)
+		return true
+	}
```

Comment:
Going back to the above discussion, storing the cluster metadata in the backup CRD spec as a nullable obejct shouldn't be a breaking change, and we can instead check if it is nil or not, rather than doing dynamic CRD reflection

IMO it is easier to make it testable if we avoid checking crd definitions and focus on the presence/absence of a new field to determine that.